main.py

```
@@ -0,0 +1,245 @@
+            rate_limit_store[client_ip] = (last_request_time, request_count + 1)
+        else:
+            rate_limit_store[client_ip] = (current_time, 1)
+    else:
+        rate_limit_store[client_ip] = (current_time, 1)
+    
+    response = await call_next(request)
+    return response
+
+@app.websocket("/ws")  # Annotation for the WebSocket endpoint
+async def websocket_endpoint(websocket: WebSocket, token: str):
+    logger.debug(f"WebSocket connection received")
```

Ruff (0.8.2)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)